sdk/tables/Azure.Data.Tables/src/TablesExceptionParser.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Data.Tables
+{
+    internal static class TablesExceptionParser
+    {
+        /// <summary>
+        /// Parses a table service exception and populates the ErrorCode and Message, if possible.
+        /// </summary>
+        /// <param name="ex"></param>
+        /// <returns></returns>
+        public static RequestFailedException Parse(RequestFailedException ex)
+        {
+            if (TryParseOdataError(ex.Message, out string code))
```

Comment:
Can you accomplish the same by overriding https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs#L45
?

Reply:
I didn't know about this! I'll use that if it works!

Reply:
We can extend it if it doesn't but that's the current way of overriding the error behavior

Reply:
This allowed me to simplify a lot!! 

![](https://media.giphy.com/media/5xtDarmwsuR9sDRObyU/giphy.gif)